Sources/PostgresNIO/New/PSQLRowStream.swift

```diff
@@ -128,10 +139,10 @@ final class PSQLRowStream {
 
     private func cancel0() {
         switch self.downstreamState {
-        case .asyncSequence(let consumer, let dataSource):
-            let error = PSQLError.connectionClosed
+        case .asyncSequence(let source, let dataSource):
+            let error = CancellationError()
```

Comment:
I don't fully agree with the usage of `CancellationError` here. It should only get thrown if the task is cancelled; however, in your code path it code be thrown when the source terminated the sequence where this would be quite confusing for the user. In general, you don't have to call `source.finish` as a result of `didTerminate` at all.